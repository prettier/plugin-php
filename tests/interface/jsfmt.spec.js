run_spec(__dirname, ["php"]);
// see https://github.com/prettier/plugin-php/issues/107
// run_spec(__dirname, ["php"], { openingBraceNewLine: false });
