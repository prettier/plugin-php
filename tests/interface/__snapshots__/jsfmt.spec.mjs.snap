// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interface.php 1`] = `
====================================options=====================================
parsers: ["php"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
<?php

interface testInterface {
    public function setVariable($test);
    public function method(array $array): iterable;
}

interface test2 extends testInterface {
  public function other($hi);
}

interface EmptyInterface {}

interface EmptyInterfaceWithComments { /* Comment */ }

interface Foo extends MyClass {}

interface Foo extends MyClass, MyOtherClass, OtherClass {}

interface Foo extends VeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongClassName {}

interface VeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongClassName extends VeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongClassName {}

interface VeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongClassName extends Foo {}

interface Foo extends MyClass, MyOtherClass, MyOtherOtherOtherClass, MyVeryVeryVeryLongClassName {}

interface VeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongClassName extends MyClass, MyOtherClass, MyOtherOtherOtherClass, MyVeryVeryVeryLongClassName {}

=====================================output=====================================
<?php

interface testInterface
{
    public function setVariable($test);
    public function method(array $array): iterable;
}

interface test2 extends testInterface
{
    public function other($hi);
}

interface EmptyInterface
{
}

interface EmptyInterfaceWithComments
{
    /* Comment */
}

interface Foo extends MyClass
{
}

interface Foo extends MyClass, MyOtherClass, OtherClass
{
}

interface Foo extends
    VeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongClassName
{
}

interface VeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongClassName
    extends VeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongClassName
{
}

interface VeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongClassName
    extends Foo
{
}

interface Foo extends
    MyClass,
    MyOtherClass,
    MyOtherOtherOtherClass,
    MyVeryVeryVeryLongClassName
{
}

interface VeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongClassName
    extends
        MyClass,
        MyOtherClass,
        MyOtherOtherOtherClass,
        MyVeryVeryVeryLongClassName
{
}

================================================================================
`;
